demos/http/http_demo_s3_download/sigv4_config.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * SigV4 Utility Library v1.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file sigv4_config.h
+ * @brief Values for configuration macros provided by the application to be
+ * used by the SigV4 Utility Library.
+ */
+
+#ifndef SIGV4_CONFIG_H_
+#define SIGV4_CONFIG_H_
+
+#ifndef SIGV4_PROCESSING_BUFFER_LENGTH
```

Comment:
It would be helpful to provide comment explanations for choice of the config values. 
For example. is it required for the Maximum Header and Query Parameter counts to be 100? As the S3 demos don't provide any query parameters, it seems like the maximum query count should be 0?